website/static/js/registrationEditorExtensions.js

```diff
@@ -0,0 +1,151 @@
+            },
+            onselectrow: function(item) {
+                if (item.kind === 'file') {
+                    viewModel.value(item.data.path);
+                    viewModel.selectedFileName(item.data.name);
+                    item.css = 'fangorn-selected';
+                } else {
+                    viewModel.value('');
+                    viewModel.selectedFileName(NO_FILE);
+                }
+            },
+            resolveRows: function(item) {
+                var tb = this;
+                item.css = '';
+
+                limitOsfStorage(item);
+
+                if (viewModel.value() !== null) {
+                    if (item.data.path === viewModel.value()) {
+                        item.css = 'fangorn-selected';
+                    }
+                }
+
+                var defaultColumns = [{
+                    data: 'name',
+                    folderIcons: true,
+                    filter: true,
+                    custom: Fangorn.DefaultColumns._fangornTitleColumn
+                }];
+                if (item.parentID) {
+                    item.data.permissions = item.data.permissions || item.parent().data.permissions;
+                    if (item.data.kind === 'folder') {
+                        item.data.accept = item.data.accept || item.parent().data.accept;
+                    }
+                }
+
+                if (item.data.uploadState && (item.data.uploadState() === 'pending' || item.data.uploadState() === 'uploading')) {
+                    return Fangorn.Utils.uploadRowTemplate.call(tb, item);
+                }
+
+                var configOption = Fangorn.Utils.resolveconfigOption.call(this, item, 'resolveRows', [item]);
+                return configOption || defaultColumns;
+            },
+            lazyLoadOnLoad: function(tree, event) {
+                limitOsfStorage(tree);
+
+                tree.children.forEach(function(item) {
+                    limitOsfStorage(item);
+
+                    if (viewModel.value() !== null) {
+                        if (item.data.path === viewModel.value()) {
+                            item.css = 'fangorn-selected';
+                            viewModel.selectedFileName(item.data.name);
+                        }
+                    }
+                    Fangorn.Utils.inheritFromParent(item, tree);
+                });
+                Fangorn.Utils.resolveconfigOption.call(this, tree, 'lazyLoadOnLoad', [tree, event]);
+                Fangorn.Utils.reapplyTooltips();
+
+                if (tree.depth > 1) {
+                    Fangorn.Utils.orderFolder.call(this, tree);
+                }
+            }
+        });
+    filePicker = fw; 
+    fw.init(); 
+    viewModel.showUploader(false);
+};
+
+ko.bindingHandlers.osfUploader = {
+    init: osfUploader
+};
+
+var Uploader = function(data) {
+
+    var self = this;
+    self._orig = data;
+
+    self.selectedFileName = ko.observable('no file selected');
```

Comment:
Can we just use the NO_FILE named constant for this text? (am touching base with @HarryRybacki to gain context on what this file and variable are used for)